java-client/src/main/java/com/cumulocity/sdk/client/ResponseParser.java

```diff
@@ -77,14 +83,29 @@ public void checkStatus(Response response, int... expectedStatusCodes) throws SD
 
     protected String getErrorMessage(Response response, int status) {
         String errorMessage = "Http status code: " + status;
+        String responseHeader = null;
 
         if (response.hasEntity()) {
             String errorRepresentation = getErrorRepresentation(response);
             if (errorRepresentation == null) {
                 errorRepresentation = NO_ERROR_REPRESENTATION;
             }
+            Map<String, List<String>> responseHeadermap = response.getStringHeaders();
+            if (responseHeadermap != null) {
+                for (Map.Entry<String, List<String>> entry : responseHeadermap.entrySet()) {
+                    responseHeader += entry.getKey() +
+                            ": " + entry.getValue() +"\n";
+                }
+            }
+//            responseHeader = response.getStringHeaders().entrySet().stream().map(e -> e.getKey() + " = " + e.getValue()).collect(Collectors.joining("&"));
             errorMessage += "\n" + errorRepresentation;
         }
+        if (platformParameters != null) {
+            errorMessage += "\n tenant: " + platformParameters.getTenantId() + " user: " + platformParameters.getUser();
```

Comment:
Personal preference (feel free to ignore): I would prefer using `String.format` to make it more readable (applies to other places too).

Reply:
Thanks for the hint. I could not find String.format neither in the class nor in the whole project, so I sticked to the original coding design ;)

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#format-java.lang.String-java.lang.Object...-